### PR TITLE
test: ShareMenuItems

### PR DESCRIPTION
**Pull request by yardz**
### SUMMARY
Tests for component ShareMenuItems

### CODE REVIEW

File: superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  copyMenuItemTitle: 'Copy dashboard URL',
+  emailMenuItemTitle: 'Share dashboard by email',
+  emailSubject: 'Superset dashboard COVID Vaccine Dashboard',
+  emailBody: 'Check out this dashboard: ',
+});
+
+const { location } = window;
+beforeAll((): void => {
+  // @ts-ignore
+  delete window.location;
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx:43]:
```suggestion
  delete window.location;
  fetchMock.post(
    'http://localhost/r/shortner/',
    { body: 'http://localhost:8088/r/3' },
    {
      sendAsJson: false,
    },
  );
```

File: superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  addDangerToast: jest.fn(),
+  addSuccessToast: jest.fn(),
+  url: '/superset/dashboard/26/?preselect_filters=%7B%7D',
+  copyMenuItemTitle: 'Copy dashboard URL',
+  emailMenuItemTitle: 'Share dashboard by email',
+  emailSubject: 'Superset dashboard COVID Vaccine Dashboard',
+  emailBody: 'Check out this dashboard: ',
+});
+
+const { location } = window;
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx:40]:
```suggestion
const { location } = window;

```

File: superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+beforeEach(() => {
+  jest.clearAllMocks();
+  fetchMock.post(
+    'http://localhost/r/shortner/',
+    { body: 'http://localhost:8088/r/3' },
+    {
+      sendAsJson: false,
+      overwriteRoutes: true,
+    },
+  );
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx:54]:
```suggestion
```

File: superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+  window.location = {
+    href: '',
+  } as any;
+});
+afterAll((): void => {
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx:59]:
```suggestion

afterAll((): void => {
```

File: superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+  emailSubject: 'Superset dashboard COVID Vaccine Dashboard',
+  emailBody: 'Check out this dashboard: ',
+});
+
+const { location } = window;
+beforeAll((): void => {
+  // @ts-ignore
+  delete window.location;
+});
+beforeEach(() => {
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx:45]:
```suggestion

beforeEach(() => {
```

File: superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+  ).toBeInTheDocument();
+  expect(
+    screen.getByRole('button', { name: 'Copy dashboard URL' }),
+  ).toBeInTheDocument();
+  expect(
+    screen.getByRole('button', { name: 'Share dashboard by email' }),
+  ).toBeInTheDocument();
+});
+
+test('Clicking on "Copy dashboard URL" and have success', async () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx:84]:
```suggestion
test('Click on "Copy dashboard URL" and succeed', async () => {
```

File: superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+  await waitFor(() => {
+    expect(spy).toBeCalledTimes(1);
+    expect(spy).toBeCalledWith('http://localhost:8088/r/3');
+    expect(props.addSuccessToast).toBeCalledTimes(1);
+    expect(props.addSuccessToast).toBeCalledWith('Copied to clipboard!');
+    expect(props.addDangerToast).toBeCalledTimes(0);
+  });
+});
+
+test('Clicking on "Copy dashboard URL" and have fail', async () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx:110]:
```suggestion
test('Click on "Copy dashboard URL" and fail', async () => {
```

File: superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+    expect(spy).toBeCalledWith('http://localhost:8088/r/3');
+    expect(props.addSuccessToast).toBeCalledTimes(0);
+    expect(props.addDangerToast).toBeCalledTimes(1);
+    expect(props.addDangerToast).toBeCalledWith(
+      'Sorry, your browser does not support copying.',
+    );
+  });
+});
+
+test('Clicking on "Share dashboard by email" and have success', async () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx:138]:
```suggestion
test('Click on "Share dashboard by email" and succeed', async () => {
```

File: superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+
+  await waitFor(() => {
+    expect(props.addDangerToast).toBeCalledTimes(0);
+    expect(window.location.href).toBe(
+      'mailto:?Subject=Superset dashboard COVID Vaccine Dashboard%20&Body=Check out this dashboard: http://localhost:8088/r/3',
+    );
+  });
+});
+
+test('Clicking on "Share dashboard by email" and have fail', async () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/menu/ShareMenuItems/ShareMenuItems.test.tsx:163]:
```suggestion
test('Click on "Share dashboard by email" and fail', async () => {
```

**Comment by junlincc** [comment]
@yardz thank you Bruno, for the PR! 

It will be great to attach the UI of each component to help testing, going forward.  cc @michael-s-molina @geido

**Review by rusackas — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**